### ISSUE TITLE
v3b2 hanging in filter federate example

### ISSUE THREAD

**Issue by trevorhardy** [GMLC-TDC/HELICS#1995]
**Describe the bug**
At some point in the filter federate example, all federates are requesting a time and none are ever granted, hanging the co-simulation.

**To Reproduce**
Run [filter federate example](https://github.com/GMLC-TDC/HELICS-Examples/tree/master/user_guide_examples/fundamental/fundamental_message_comm/filter_federate) and look at the logs to confirm all federates have requested a time and are waiting to be granted one.

**Environment (please complete the following information):**

- Operating System: Mac
- Language Extension: Python
- what compiler or setup process did you use: download v3b2 and install
- HELICS version: v3b2 (2021-06-30)

**Additional context and information**
I retested this on my installation and this is failing under v3b1 as well though I thought it was previously working under that release.

**Comment by trevorhardy** [comment]
v3.1 update:

When setting "event_triggered" undefined the co-simulation hangs.
When setting "event_triggered" to "true" and "grant_timeout" to "0" the co-simulation runs to completion. 
When setting "event_triggered" to "true" and "grant_timeout" to "1"  the co-simulation runs to completion **BUT** generates a different result than when "grant_timeout" is "0".


**With grant_timeout = 0:**
![Screen Shot 2021-12-06 at 11 27 30 AM](https://user-images.githubusercontent.com/8697189/144910625-07731f86-5905-47d8-8d8b-fdf26f53e485.png)


**With grant_timeout = 1:**
![Screen Shot 2021-12-06 at 11 26 29 AM](https://user-images.githubusercontent.com/8697189/144910528-6bdd7253-fac3-41c5-8fb0-ce1037c4c990.png)

Progress?

**Comment by trevorhardy** [comment]
Checked on v3.2.0 and federation is still hanging.

**Comment by phlptp** [comment]
@trevorhardy  can you check on 3.2.1?

**Comment by trevorhardy** [comment]
Check on v3.2.1 and federation is still hanging.

**Comment by phlptp** [comment]
@trevorhardy  you want to test this again with the latest release (or 3.4 when it comes out)

**Comment by trevorhardy** [comment]
Yes. I can't remember whether I tested this in v3.3 and I'll be sure to do so under v3.4.

**Comment by trevorhardy** [comment]
Federation still hangs in v3.6.0. All federates requesting a time and none being granted one.

**Comment by trevorhardy** [comment]
Test on dev commit f347e38fe81e8e7d858a0e2255f2859118a06dc4 and still hangs.
